symphony-ws/src/main/java/se/havochvatten/symphony/service/ReportService.java

```diff
@@ -420,4 +415,290 @@ static void writeComparisonRows(Locale locale, SymphonyBandDto[] meta, int[] fea
+                ComparisonResultExport cmpResult = new ComparisonResultExport();
+                cmpResult.calculationName = cmpResults[i].calculationName;
+                cmpResult.ecosystemTitles = Arrays.stream(layersToList[i].ecosystems)
+                    .mapToObj(e -> ecoTitles.get(cmpResults[i].includedEcosystems[e]))
+                    .toArray(String[]::new);
+                cmpResult.pressureTitles = Arrays.stream(layersToList[i].pressures)
+                    .mapToObj(p -> pressureTitles.get(cmpResults[i].includedPressures[p]))
+                    .toArray(String[]::new);
+                cmpResult.comparisonMatrix = layersToList[i].result;
+                cmpResult.cumulativeTotal = cmpResults[i].cumulativeTotal;
+                return cmpResult;
+
+            }).toArray(ComparisonResultExport[]::new);
+
+        return result;
+    }
+
+    private OdfTable createSheet(OdfSpreadsheetDocument doc, int rows, int cols, String title) {
+        OdfTable newSheet = OdfTable.newTable(doc, rows, Math.min(cols, 5));
+
+        newSheet.getCellRangeByPosition(2, 1, 5, 1).merge();
+        OdfTableCell titleCell = newSheet.getCellByPosition(2, 1);
+        titleCell.setStringValue(title);
+        titleCell.getOdfElement().setStyleName("cmpName");
+
+        newSheet.getColumnByIndex(0).setWidth(5);
+        newSheet.getRowByIndex(1).setHeight(14, false);
+
+        return newSheet;
+    }
+
+    public byte[] generateMultiComparisonAsODS(
+        CompoundComparison comparison, String preferredLanguage, String localisedHeading, boolean excludeZeroes) throws Exception {
+
+        String title = localisedHeading + " \"" + comparison.getName() + "\"";
+
+        Map<Integer, String>
+            ecoTitles = metaDataService.getComponentTitles(
+                comparison.getBaseline().getId(),
+                LayerType.ECOSYSTEM, preferredLanguage),
+            pressureTitles = metaDataService.getComponentTitles(
+                comparison.getBaseline().getId(),
+                LayerType.PRESSURE, preferredLanguage);
+
+        // The ODF template file 'template.ods' that is packaged with the application is prepared
+        // with the following named office styles (in "styles.xml") :
+        // "cmpName", "calcName", "ecoHeader", "pressureHeader", "value", "totalHE", "totalHP",
+        // "totalE", "totalP", "totalC", "resultSep"
+
+        OdfPackage odfPackage = loadPackage(
+            getClass().getClassLoader().getResourceAsStream("document/template.ods")
+        );
+
+        OdfSpreadsheetDocument templateDocument = (OdfSpreadsheetDocument) odfPackage.getRootDocument();
+
+        ComparisonResult[] cmpResults =
+            excludeZeroes ?
+                comparison.getResult().values().stream()
+                    .filter(cmp -> cmp.cumulativeTotal != 0)
+                    .toArray(ComparisonResult[]::new) :
+            comparison.getResult().values().toArray(new ComparisonResult[0]);
```

Comment:
Should not this also create a `ComparisonResult[]::new`?

Reply:
Not sure what you mean. It's the common Stream API syntax for converting to array (for objects).